server/models/ticket.js

```diff
@@ -40,7 +67,43 @@ TicketSchema.statics = {
         reject(error);
       });
     });
+  },
+
+  addAction(id, action, timestamp, userId) {
```

Comment:
`userId` is the only optional parameter here, there are cases when there will not be a user attached to an action (created for example).

I have timestamp as a required field, so you're right that it shouldn't have the `|| <date>`. It's required because if there is some delay in processing, we have the original time that the action was initiated. This seems like it's worth talking about further.